compat/win32/trace2_win32_process_info.c

```diff
@@ -38,16 +44,28 @@ static void get_processes(struct json_writer *jw, HANDLE hSnapshot)
 {
 	PROCESSENTRY32 pe32;
 	DWORD pid;
+	DWORD pid_list[NR_PIDS_LIMIT];
+	int k, nr_pids = 0;
 
 	pid = GetCurrentProcessId();
+	while (find_pid(pid, hSnapshot, &pe32)) {
+		/* Only report parents. Omit self from the JSON output. */
+		if (nr_pids)
+			jw_array_string(jw, pe32.szExeFile);
 
-	/* We only want parent processes, so skip self. */
-	if (!find_pid(pid, hSnapshot, &pe32))
-		return;
-	pid = pe32.th32ParentProcessID;
+		/* Check for cycle in snapshot. (Yes, it happened.) */
```

Comment:
Do we need to check for cycles or should we just give up once we exceed NR_PIDS_LIMIT?

Reply:
yea, i could go either way here.  my way lets us see "(cycle)" vs "(truncate)" in the Kusto data.
i thought that might be interesting.

Reply:
If we check for a cycle (which we know happens), we will avoid having `NR_PIDS_LIMIT` lines of repeated entries every time it does happen...